example-get-started-cv/generate.sh

```diff
@@ -0,0 +1,197 @@
+#!/bin/bash
+# See https://dvc.org/doc/tutorials/get-started
+
+# Setup script env:
+#   e   Exit immediately if a command exits with a non-zero exit status.
+#   u   Treat unset variables as an error when substituting.
+#   x   Print commands and their arguments as they are executed.
+set -eux
+
+HERE="$( cd "$(dirname "$0")" ; pwd -P )"
+REPO_NAME="example-get-started-cv"
+REPO_PATH="$HERE/build/$REPO_NAME"
+PROD=${1:-false}
+
+if [ -d "$REPO_PATH" ]; then
+  echo "Repo $REPO_PATH already exists, please remove it first."
+  exit 1
+fi
+
+TOTAL_TAGS=15
+STEP_TIME=100000
+BEGIN_TIME=$(( $(date +%s) - ( ${TOTAL_TAGS} * ${STEP_TIME}) ))
+export TAG_TIME=${BEGIN_TIME}
+export GIT_AUTHOR_DATE="${TAG_TIME} +0000"
+export GIT_COMMITTER_DATE="${TAG_TIME} +0000"
+tick(){
+  export TAG_TIME=$(( ${TAG_TIME} + ${STEP_TIME} ))
+  export GIT_AUTHOR_DATE="${TAG_TIME} +0000"
+  export GIT_COMMITTER_DATE="${TAG_TIME} +0000"
+}
+
+export GIT_AUTHOR_NAME="Alex Kim"
+export GIT_AUTHOR_EMAIL="alex000kim@gmail.com"
+export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"
+export GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"
+
+mkdir -p $REPO_PATH
+pushd $REPO_PATH
+
+virtualenv -p python3 .venv
+export VIRTUAL_ENV_DISABLE_PROMPT=true
+source .venv/bin/activate
+echo '.venv/' > .gitignore
+
+# Installing from main since we'd like to update repo before
+# the release
+pip install "git+https://github.com/iterative/dvc#egg=dvc[all]"
+
+git init
+cp $HERE/code/README.md .
+cp $HERE/code/.devcontainer.json .
+cp $HERE/code/.gitattributes .
+cp $HERE/code/requirements.txt .
+
+git add .
+tick
+git commit -m "Initialize Git repository"
+git tag -a "0-git-init" -m "Git initialized."
+
+dvc init
+tick
+git commit -m "Initialize DVC project"
+
+cp -r $HERE/code/data .
+git add data/.gitignore data/pool_data.dvc
+tick
+git commit -m "Add data"
+dvc pull
+
+# Remote active on this env only, for writing to HTTP redirect below.
+dvc remote add -d --local storage s3://dvc-public/remote/get-started-pools
```

Comment:
@shcheklein Could you set up some repo (& codespace) secrets for accessing `s3://dvc-public/` ? 

This generate.sh is kind of heavy and would be nice to run it from a codespace with GPU here

Reply:
you both should have access to GPU codespaces right? We can run this repo, I don't think we need a separate one, right? I can create a devcontainer here ...

Besides that:

- can it be run on Mac with GPU support (ideally we should be able to run it). It means that hopefully not everyone needs this
- we have a ticket to simplify Codespaces -> S3 interaction, but we haven't done it yet. Let's for make sure that sandbox credentials are enough to write to `dvc-public` and we could use those then inside codespaces. Would that work?

Reply:
IIRC, `s3://dvc-public/` is not under the sandbox account. 
> you both should have access to GPU codespaces right?` 
Right, but every time I launch codespaces, I need to 1) generate my personal temporary MFA creds
and 2) set them up in the codespaces instance.

Reply:
> IIRC, s3://dvc-public/ is not under the sandbox account.

Yes, but it has permissions set so that any sandbox user can write / read to /from it. Could you please check that?

> Right, but every time I launch codespaces, I need to 1) generate my personal temporary MFA creds
and 2) set them up in the codespaces instance.

Looking into what @mjasion recommended here https://github.com/iterative/iterative.ai/issues/555#issuecomment-1358027948 - I think it should be possible to setup it ... I'll try to create a dev container for this repo.

Reply:
>Yes, but it has permissions set so that any sandbox user can write / read to /from it. Could you please check that?

Reading works, writing doesn't.
```
$ aws s3 ls s3://dvc-public/remote/get-started-pools
                           PRE get-started-pools/
$ aws s3 cp empty_test_file.txt s3://dvc-public/remote/test-dir/
upload failed: ./empty_test_file.txt to s3://dvc-public/remote/test-dir/empty_test_file.txt An error occurred (AccessDenied) when calling the PutObject operation: Access Denied
```

Reply:
> can it be run on Mac with GPU support (ideally we should be able to run it). It means that hopefully not everyone needs this

Trying to run the notebook on my m1 laptop as is runs but is very slow (I think it will take hours to finish) and makes the whole machine laggy.

Reply:
>  we have a ticket to simplify Codespaces -> S3 interaction, but we haven't done it yet. Let's for make sure that sandbox credentials are enough to write to dvc-public and we could use those then inside codespaces. Would that work?

Yes, I confirm that I can write from codespace with sandbox account, +1 to simplify the process, though, setting up the sandbox + MFA feels inconvenient.

A different problem I have found inside the codespaces is the lack of write permissions on the destination repo (in order to do the force pushes from the codespace). 
I have tried https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces but didn't seem to work

Reply:
Okay, you should be able now to run Codespaces for this repo. @dberenbaum @daavoo @alex000kim you folks have access to S3 w/o MFA, including `s3://dvc-public`. Please give it a try.

I'm looking into repos permissions ...

Reply:
I've added permission per https://github.com/iterative/example-repos-dev/commit/2e11e66276e6a2f41f0bc2bd0ab422d4c5d6e711 .

Reply:
Thanks @shcheklein !

Reply:
@daavoo @alex000kim It works for me, I've generated and force-pushed the repo from Codespaces.

Reply:
For the record, everything works for me. 

> I have tried https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces but didn't seem to work

It turned out that I made a typo when creating the `.devcontainer` file 🤦 that's why it didn't work